ls1intum/Artemis
# Feature/grading instructions/modify markdown editor

### Checklist

### Motivation and Context


Given the current situation, instructors add a block of text describing the grading instructions in the text editor. However, in order to unify the assessment feedback, we have to divide the text block in specific parameters, that could, later on, be dragged and dropped by tutors when assessing an submission.
### Description

Added the following domain commands :
- Credits
- Grading Scale
- Grading Instruction
- Feedback
- Usage Count (which indicates how often the points should be applied for the current instruction)

to the text editor of the grading instructions and by that we divided the text block into four text fields. The above-discribed commands are all triggered by the `GradingCriteriaCommand` to enable users to add all parameters of instruction via one click.

### Steps for Testing


1. Navigate to Course Administration
2. Navigate to Exercise
3. Edit Modeling/text/file-upload exercises
4. Add free text in the grading instructions editor and save it
5. Switch to structured Grading instructions -> if you switch back to grading instructions the free text inserted before should still be available
6. Add a new criterion/ edit existing criterion/ delete an existing criterion
7. Add one or multiple instructions to a criterion of your choice/ edit and delete it
8. Save your changes -> when you click on edit again your changes should be displayed correctly

Note: it is not possible to delete a single instruction. Only a criterion can be deleted and all its corresponding instructions will therefore also be removed. This limitation will be addressed in a separate PR

### Screenshots


![Recording-3](https://user-images.githubusercontent.com/50635005/75719920-e9e35280-5cd5-11ea-8788-7f95d817d13c.gif)
